Assignment4/u120182 - SPARQL.txt

```diff
@@ -0,0 +1,83 @@
+BASIC
+
+1.
+
+select distinct ?x where {[] a ?x} LIMIT 100
+
+2.
+
+select distinct ?a where {?x rdfs:subClassOf <http://GP-onto.fi.upm.es/exercise2#Establishment> . ?x <http://www.w3.org/2000/01/rdf-schema#label> ?a}
+
+3.
+
+select distinct ?y where {?x <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://GP-onto.fi.upm.es/exercise2#City>. ?x rdfs:label ?y}
+
+4.
+
+select distinct ?x where {<http://GP-onto.fi.upm.es/exercise2#Santiago_de_Compostela> <http://GP-onto.fi.upm.es/exercise2#hasInhabitantNumber> ?x}
+
+5.
+
+select distinct ?x ?y where { <http://GP-onto.fi.upm.es/exercise2#Santiago_de_Compostela> <http://GP-onto.fi.upm.es/exercise2#hasInhabitantNumber> ?x . <http://GP-onto.fi.upm.es/exercise2#Arzua> <http://GP-onto.fi.upm.es/exercise2#hasInhabitantNumber> ?y}
+
+6.
+
+select distinct ?z ?y where {?x <http://GP-onto.fi.upm.es/exercise2#hasInhabitantNumber> ?y . ?x rdfs:label ?z.} ORDER BY ASC(?z)
+
+7.
+
+select distinct ?a ?b where {?x <http://www.w3.org/2000/01/rdf-schema#subClassOf> <http://GP-onto.fi.upm.es/exercise2#Locality>. ?z ?y ?x . ?z <http://www.w3.org/2000/01/rdf-schema#label> ?a. OPTIONAL {?z <http://GP-onto.fi.upm.es/exercise2#hasInhabitantNumber> ?b}}
+ 
+8.
+ 
+select distinct ?a ?b where {?x <http://www.w3.org/2000/01/rdf-schema#subClassOf> <http://GP-onto.fi.upm.es/exercise2#Locality>. ?z ?y ?x . ?z <http://www.w3.org/2000/01/rdf-schema#label> ?a. ?z <http://GP-onto.fi.upm.es/exercise2#hasInhabitantNumber> ?b. FILTER (xsd:integer(?b) > xsd:integer("200000"))}
+
```

Comment:
Not necessary to have the xsd:integer("200000") if you just write 200000, but still working.